python-spec/src/somacore/query/query.py

```diff
@@ -587,11 +588,13 @@ def _var_df(self) -> data.DataFrame:
 
     @property
     def _threadpool(self) -> futures.ThreadPoolExecutor:
-        """Creates a thread pool just in time."""
+        """
+        Returns the threadpool provided by the experiment's context.
+        If not available, creates a thread pool just in time."""
+        if self.experiment.context._threadpool is not None:
```

Comment:
the Experiment-ish protocol types _threadpool as a required value, i.e., won't be None.  Which is correct?

Reply:
See comment above - I put all of them as `Optional`.